front-end/hot-dog/src/CustomerComponents/CustomerMenu.js

```diff
@@ -7,12 +7,34 @@ class CustomerMenu extends React.Component{
         this.state = {tableDataisFetched: false};
     }
 
+    submitOrder = () => {
+        // create JSON object for back-end API
+        let order = {}
+        for(let key in this.state){
+            order[key] = this.state[key]
+        }
+        let orderJSON = {
+            "order": order,
+            "cartID":this.props.cartID,
+            "userID":2
+        }
+
+        fetch(`http://localhost:5000/customer/order/`, {
```

Comment:
To send a JSON as the body of the request, you would need to specify a header like `headers: { 'Content-Type': 'application/json' }` **and** stringify the JSON object. These two reasons are probably what's causing the `undefined` behavior in the back-end.

Example [line 50-53]: https://github.com/eknigge/AD320_Project/blob/main/front-end/hot-dog/src/components/TableRow.js

Source: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_json_data